lib/src/rules/avoid_library_directive.dart

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) 2015, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+
+import '../analyzer.dart';
+
+const _desc =
+    'Avoid library directives unless that have documentation comments or '
+    'annotations';
+
+const _details = r'''
+**DO** use library directives if you want to document a library and/or annotate 
+a library.
+
+**GOOD:**
+```dart
+/// This library does important things
+library;
+```
+
+```dart
+@TestOn('js')
+library;
+```
+
+**BAD:**
+```dart
+library;
+```
+''';
+
+class AvoidLibraryDirective extends LintRule {
+  static const LintCode code = LintCode('avoid_library_directive',
+      'Library directives without comments or annotations should be avoided.',
+      correctionMessage: 'Try deleting the library directive.');
+
+  AvoidLibraryDirective()
+      : super(
+            name: 'avoid_library_directive',
```

Comment:
Yes, we want to not use `avoid` whenever possible. Consider `unnecessary_library_directive` instead.

Reply:
Done!